openlibrary/plugins/upstream/code.py

```diff
@@ -88,11 +88,15 @@ def GET(self):
 
 
 class merge_work(delegate.page):
-    path = "(/works/OL\d+W)/merge"
-    def GET(self, key):
-        return "This looks like a good place for a merge UI!"
+    path = "/works/merge"
 
-    def POST(self, key):
+    def GET(self):
+        user = web.ctx.site.get_user()
+        if not user or (not user.is_admin() and not user.is_librarian()):
+            raise web.HTTPError('403 Forbidden')
+        return render_template('merge/works')
```

Comment:
Too much negativity...
```suggestion
        if user and (user.is_admin() or user.is_librarian()):
            return render_template('merge/works')
        raise web.HTTPError('403 Forbidden')
```

Reply:
seems like a reasonable proposal to me; the default is to fail and only succeed if it meets our specific criteria

Reply:
:+1: Fixed